### PR TITLE
Fixes Interpolation in SimpleATsit5

### PR DESCRIPTION
**Pull request by kanav99**
Fixes issue #18 by adding a new field `dtnew` in the cache

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Should add a test on the interpolation.

**Comment by Datseris** [comment]
Yeah something along the lines:
```
uprev = copy(integ.u)
step!(integ)

@test uprev \approx integ(integ.tprev)
@test integ(integ.t) \approx integ.u
```

**Comment by ChrisRackauckas** [comment]
and test against Tsit5.

**Comment by kanav99** [comment]
Tests really do fail. I checked with the other way around by doing `dt = tnext - tprev` and the tests pass then. I will investigate this now.

**Comment by Datseris** [comment]
Tests seem to pass! I'll test it as well now!

**Comment by Datseris** [comment]
@kanav99 unfortunatelly this does not solve the problem. Just use the code I use in : https://github.com/JuliaDiffEq/SimpleDiffEq.jl/issues/18#issue-408097433 

You will see that even in this PR the same problem happens: Tsit5 correctly finds the PSOS but SImpleATsit5 does not.

I think before we do anything else, you first have to be sure that this is checked:
```
uprev = copy(integ.u)
step!(integ)

@test uprev \approx integ(integ.tprev) atol = 1e-12
@test integ(integ.t) \approx integ.u atol = 1e-12
```

(which is not checked in the current tests)

**Comment by kanav99** [comment]
Added the interpolation test. This runs fine. So the problem we have now is that it doesn't match against Tsit5(). I will continue to investigate.

**Comment by Datseris** [comment]
@kanav99 I do not know what is going on, but the interpolation tests do not pass locally on my machine:
```
julia> uprev = copy(oop.u)
3-element SArray{Tuple{3},Float64,1,3}:
 10.0
 10.0
 10.0

julia> 

julia> step!(oop)

julia> 

julia> @test uprev ≈ oop(oop.tprev) atol = 1e-12
Test Passed

julia> 

julia> @test oop(oop.t) ≈ oop.u atol = 1e-12
Test Failed at none:1
  Expression: ≈(oop(oop.t), oop.u, atol=1.0e-12)
   Evaluated: [10.0389, 10.7752, 10.3801] ≈ [10.0182, 10.7862, 10.3599] (atol=1.0e-12)
ERROR: There was an error during testing
```

Specifically, as I have said in the very beginning of this issue, it is this: `integ(integ.t) != integ.u` which is the problem. If we can solve this then have have solved the issue.

**Comment by Datseris** [comment]
If we can solve up to Tuesday that would be amazing, otherwise I have to revert the default solver change :(

**Comment by Datseris** [comment]
Maaaan I truly don't understand what is going on. How can these tests pass on Travis and not on my local machine??? Could it _really_ be that I am on Windows? I cannot believe it.

I am checked out in the branch `kanav99/atsit5interp` and when I run them I get the error I posted above. :S :S :S

**Comment by kanav99** [comment]
There is surely some problem in the comparision between `Tsit5()` and `SimpleATsit5()` because even the results of the normal steps (not interpolation) vary by a great deal.
But the above tests seem to pass on my machine without any problem. Can you check whats the last commit on your branch using `git show` and see if its my commit?

**Comment by Datseris** [comment]
It's 580371761b0b49fd5a2adf73d7ce66ed0b01fb5b according to gitkraken.

File: src/tsit5/atsit5.jl [hunk 1]
```diff
@@ -186,18 +187,18 @@ function DiffEqBase.step!(integ::SAT5I{true, S, T}) where {S, T}
...
-        dtold = dt
-        dt = dt/q #dtnew
-        dt = min(abs(dt),abs(tf-t-dtold))
+        dtnew = dt/q #dtnew
+        dtnew = min(abs(dtnew),abs(tf-t-dt))
 
         integ.dt = dt
+        integ.dtnew = dtnew
         integ.qold = qold
         integ.tprev = t
```
Comment (ChrisRackauckas) [inline src/tsit5/atsit5.jl]: this is moving the interval too soon

**Pull request merged**